File: 1주차/SWEA1232. [SW 문제해결 기본] 9일차 - 사칙연산/백동승.py
def checkIsOp(data):
    if data in "+-*/": return True
    else: return False
         
class Node:
    def __init__(self, isOp, data, left, right):
        self.isOp = isOp
        self.data = data
        self.left = int(left)
        self.right =int(right)
         
    def cal(self):
        if not self.isOp: return int(self.data)
         
        if self.data == "+": return  nodes[self.left].cal() + nodes[self.right].cal()
        elif self.data == "-": return  nodes[self.left].cal() - nodes[self.right].cal()
        elif self.data == "*": return  nodes[self.left].cal() * nodes[self.right].cal()
        elif self.data == "/": return  nodes[self.left].cal() / nodes[self.right].cal()
 
for test_case in range(1,11):
    n = int(input())
    nodes = [-1]
    for i in range(n):
        line = input().split()
        isOp = checkIsOp(line[1])
        if(isOp): nodes.append(Node(isOp, line[1], line[2], line[3]))
        else: nodes.append(Node(isOp, line[1], 0, 0))
             
    print("#{} {}".format(test_case, int(nodes[1].cal())))

File: 1주차/SWEA1232. [SW 문제해결 기본] 9일차 - 사칙연산/백동승.java
import java.io.*;
 
class Solution
{
    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    private static Node[] nodes;
    
    private static boolean checkIsOp(char data){
        return (data == '+' || data == '-' || data == '*' || data == '/');
    }
     
    private static class Node{
        public boolean isOp;
        public char op;
        public double num;
        public int left, right;
        Node(boolean isOp, char op, double num, int left, int right){
            this.isOp = isOp;
            this.op = op;
            this.num = num;
            this.left = left;
            this.right = right;
        }
         
        public double cal(){
            if(!isOp) return num;
             
            switch(this.op){
                case '+': return nodes[left].cal() + nodes[right].cal();
                case '-': return nodes[left].cal() - nodes[right].cal();
                case '*': return nodes[left].cal() * nodes[right].cal();
                case '/': return nodes[left].cal() / nodes[right].cal();
            }
            return 0;
        }
         
    }
     
    public static void main(String args[]) throws Exception
    {
 
        for(int test_case = 1; test_case <= 10; test_case++)
        {
            int n = Integer.parseInt(br.readLine());
            nodes = new Node[n+1];
            for(int i = 0; i < n; i++){
                String[] line = br.readLine().split(" ");
                int idx = Integer.parseInt(line[0]);
                boolean isOp = checkIsOp(line[1].charAt(0));
                 
                if(isOp){
                    char op = line[1].charAt(0);
                    int left = Integer.parseInt(line[2]);
                    int right = Integer.parseInt(line[3]);
                    nodes[idx] = new Node(isOp, op, 0, left, right);
                }
                else nodes[idx] = new Node(isOp, '_', Double.valueOf(line[1]), 0, 0);
            }
            bw.write("#"+test_case+" "+Math.round(nodes[1].cal())+"\n");
        }
        bw.flush();
        br.close();
        bw.close();
    }
}
